EIPS/eip-5453.md

```diff
@@ -0,0 +1,170 @@
+---
+eip: 5453
+title: Smart Endorsement
+description: Defines a format for endorsement on-chain
+author: Zainan Victor Zhou (@xinbenlv)
+discussions-to: https://ethereum-magicians.org/t/erc-5453-endorsement-standard/10355
+status: Draft
+type: Standards Track
+category: Interface
+created: 2022-08-12
+requires: 165
+---
+<!-- TODO add 5269 in requires after the ./eip-5269.md becomes available. -->
+## Abstract
+
+Provides a format for endorsement of smart contract transactions in the format of extra data (the last bytes field of a method signature).
+
+## Motivation
+
+1. Support a second approval from another user.
+2. Support pay-for-by another user
+3. Support multi-sig
+4. Support persons acting in concert by endorsements
+5. Support accumulated voting
+6. Support off-line signatures
+
+TODO add implementation references for each of the case above.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+1. A field of method signature MUST be designated to be the endorsement field.
+It must exceed a minimum length of `65 bytes` which is the length of (v, r, s) where as
+
+- `r, s` are the ECDSA recover signature pair. `v`'s last bit carries ECDSA recover Y-Parity.
+
+2. For any complying method of compliant contract, the last data field MUST be `bytes`.
+
+3. Such data field MUST conform to the following format
+
+3.1. the last 8-bytes MUST be a magic 8-bytes word in the ASCII representation of "ENDORSED".
+Implementing method of smart contract that honors smart endorsement MUST deemed a transaction unendorsed by anyone,
+if the ending 8-bytes doesn't matches with this magic word.
+3.2. the last 2-bytes MUST be a type-indicator, denoted as `Erc5453FormatType`
+3.3. For `Erc5453FormatType == FIRST_TYPE_SINGLE_ENDORSER` <!--TODO determine a type number-->,
+it supports a simple endorsement signature in the following format. Any complying contract MUST implement FIRST_TYPE.
+Other type number will be used for future extension, such as a multi-but-separate-endorsement scenario.
+
+3.4. Complying contract MUST emit the following `OnEndorsed` event when a transaction is executed because of the endorsement. Complying contract MUST NOT emit such event when a transaction is executed without taking account of the endorsement.
```

Comment:
```suggestion
  1. the last 8-bytes MUST be a magic 8-bytes word in the ASCII representation of "ENDORSED". Implementing method of smart contract that honors smart endorsement MUST deemed a transaction unendorsed by anyone,
if the ending 8-bytes doesn't matches with this magic word.
  2. the last 2-bytes MUST be a type-indicator, denoted as `Erc5453FormatType`
  3. For `Erc5453FormatType == FIRST_TYPE_SINGLE_ENDORSER`, it supports a simple endorsement signature in the following format. Any complying contract MUST implement FIRST_TYPE. Other type number will be used for future extension, such as a multi-but-separate-endorsement scenario.
  4. Complying contract MUST emit the following `OnEndorsed` event when a transaction is executed because of the endorsement. Complying contract MUST NOT emit such event when a transaction is executed without taking account of the endorsement.
```